packages/account/src/Sections/Security/TwoFactorAuthentication/__tests__/digit-form.spec.tsx

```diff
@@ -0,0 +1,150 @@
+        });
+    });
+
+    it('should display error if user types less than 6 digits', async () => {
+        render(<DigitForm {...mock_props} />);
+        const digitInput = screen.getByRole('textbox');
+
+        userEvent.type(digitInput, '6699');
+
+        await waitFor(() => {
+            const error = screen.getByText(/Length of digit code must be 6 characters./i);
+            expect(error).toBeInTheDocument();
+        });
+    });
+
+    it('should display error if user types invalid OTP', async () => {
+        render(<DigitForm {...mock_props} />);
+        const digitInput = screen.getByRole('textbox');
+        const invalidOTP = '786789';
+        userEvent.type(digitInput, invalidOTP);
+
+        const submitButton = screen.getByRole('button', { name: /Enable/i });
+        userEvent.click(submitButton);
+
+        await waitFor(() => {
+            const error = screen.getByText(/That's not the right code. Please try again./i);
+            expect(error).toBeInTheDocument();
+        });
+    });
+
+    it('should disable button when form is empty or validation fails', async () => {
+        render(<DigitForm {...mock_props} />);
+        const submitButton = screen.getByRole('button', { name: /Enable/i });
+        expect(submitButton).toBeDisabled();
+
+        const digitInput = screen.getByRole('textbox');
+        userEvent.type(digitInput, '669yi9');
+        await waitFor(() => {
+            expect(submitButton).toBeDisabled();
+        });
+    });
+
+    it('should display error if response error object does not contain error code', async () => {
+        WS.authorized.accountSecurity.mockResolvedValue({
+            error: { message: '', code: '' },
+        });
+
+        render(<DigitForm {...mock_props} />);
+
+        const digitInput = screen.getByRole('textbox');
+        const invalidOTP = '786789';
+
+        userEvent.type(digitInput, invalidOTP);
+
+        const submitButton = screen.getByRole('button', { name: /Enable/i });
+        userEvent.click(submitButton);
+
+        await waitFor(async () => {
+            const error = screen.queryByText(/That's not the right code. Please try again./i);
+            expect(error).not.toBeInTheDocument();
+        });
+    });
+
+    it('should display error if response error object does not contain error code', async () => {
+        WS.authorized.accountSecurity.mockResolvedValue({
+            success: { message: '', code: '' },
+        });
+
+        render(<DigitForm {...mock_props} />);
+
+        const digitInput = screen.getByRole('textbox');
+        const invalidOTP = '786789';
+
+        userEvent.type(digitInput, invalidOTP);
+
+        const submitButton = screen.getByRole('button', { name: /Enable/i });
+        userEvent.click(submitButton);
+
+        await waitFor(async () => {
+            const error = screen.queryByText(/That's not the right code. Please try again./i);
+            expect(error).not.toBeInTheDocument();
+        });
```

Comment:
The description says "should display error" , but we are checking error not to be in the document.?

Reply:
My bad, the description should have been `should not display error if response object does not contain error object`